src/Donors/Controllers/DonorRequestController.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Give\Donors\Controllers;
+
+use Give\Donations\ValueObjects\DonationMetaKeys;
+use Give\Donors\Models\Donor;
+use Give\Donors\Repositories\DonorRepository;
+use Give\Donors\ValueObjects\DonorRoute;
+use Give\Framework\QueryBuilder\JoinQueryBuilder;
+use Give\Framework\QueryBuilder\QueryBuilder;
+use WP_Error;
+use WP_REST_Request;
+use WP_REST_Response;
+
+/**
+ * @unreleased
+ */
+class DonorRequestController
+{
+    /**
+     * @unreleased
+     *
+     * @return WP_Error | WP_REST_Response
+     */
+    public function getDonor(WP_REST_Request $request)
+    {
+        $donation = Donor::find($request->get_param('id'));
+
+        if ( ! $donation) {
+            return new WP_Error('donor_not_found', __('Donor not found', 'give'), ['status' => 404]);
+        }
+
+        return new WP_REST_Response($donation->toArray());
+    }
+
+    /**
+     * @unreleased
+     */
+    public function getDonors(WP_REST_Request $request): WP_REST_Response
+    {
+        $page = $request->get_param('page');
+        $perPage = $request->get_param('per_page');
+
+        $query = give(DonorRepository::class)->prepareQuery();
+
+        if ($campaignId = $request->get_param('campaignId')) {
+            $query->select(['donationmeta1.donation_id', 'donationId'])
+                ->distinct()
+                ->join(function (JoinQueryBuilder $builder) use ($campaignId) {
+                    $builder->innerJoin('give_donationmeta', 'donationmeta1')
+                        ->joinRaw("ON donationmeta1.meta_key = '" . DonationMetaKeys::DONOR_ID . "' AND donationmeta1.meta_value = ID");
+
+                    $builder->innerJoin('give_donationmeta', 'donationmeta2')
+                        ->joinRaw("ON donationmeta2.meta_key = '" . DonationMetaKeys::CAMPAIGN_ID . "' AND donationmeta2.meta_value = {$campaignId}");
+                })->whereIn('donationmeta1.donation_id', function (QueryBuilder $builder) {
+                    $builder
+                        ->select('ID')
+                        ->from('posts')
+                        ->whereRaw("WHERE ID = donationmeta1.donation_id AND post_type = 'give_payment' AND post_status = 'publish'");
+                });
+        }
+
+        $query
+            ->limit($perPage)
+            ->offset(($page - 1) * $perPage);
+
+        $donors = $query->getAll() ?? [];
```

Comment:
Does this need to account for Anonymous Donations? Should those be removed or redacted?

Reply:
Btw, in the Donors Block, we have a “Show Anonymous” setting where the admin can decide whether to show or hide anonymous donations. That might be an argument for this route. Additionally, we might have the opportunity to set an `orderby` argument, as we’ll need to retrieve the top donors based on the amount they donated.

Reply:
@kjohnson I added a new `hideAnonymousDonations` param here: 3ada42b77054d82363f10c2459ea87d321135567

@pauloiankoski The orderBy still needs to be implemented.

Reply:
After my last comment, I realized that you folks are commenting in the `DonorRequestController` class and I implemented the new param on the `DonationRequestController` class. However, I'm working to implement a similar approach in the Donor controller as well. I'll comment here when it's done.

Reply:
Now is possible to exclude anonymous donations/donors and also order by the results using different columns and and directions.

Since I did a bunch of commits with these changes, please re-check the PR description because I included all the details about these changes.